firebase-firestore/src/main/java/com/google/firebase/firestore/FirestoreMultiDbComponent.java

```diff
@@ -41,16 +48,39 @@
     this.context = context;
     this.app = app;
     this.authProvider = authProvider;
+    this.app.addLifecycleEventListener(this);
```

Comment:
Just a consideration: `app.delete()` is not a documented API and no other SDK integrates with it(hence it is effectively almost a noop), I'll defer to your judgement on whether you want to integrate with it

Reply:
sgtm